index.js

```diff
@@ -354,3 +354,114 @@ async function pickRandomFileWithCDN({
     return { rootCid, setId, rootId, clientAddress }
   }
 }
+
+/**
+ * @param {object} args
+ * @param {PdpVerifier} args.pdpVerifier
+ * @param {PandoraService} args.pandoraService
+ * @param {string} args.CDN_HOSTNAME
+ */
+
+export async function testLatestRetrievableRoot({
+  pdpVerifier,
+  pandoraService,
+  CDN_HOSTNAME
+}) {
+  const { rootCid, proofSetId, rootId, clientAddress } = await getMostRecentFileWithCDN(
+    {
+      pdpVerifier,
+      pandoraService
+    },
+  )
+
+  await testRetrieval({
+    pdpVerifier,
+    pandoraService,
+    clientAddress,
+    CDN_HOSTNAME,
+    rootCid,
+    setId: proofSetId,
+    rootId,
+  })
+}
+
+/**
+ * @param {Object} args
+ * @param {PdpVerifier} args.pdpVerifier
+ * @param {PandoraService} args.pandoraService
+ * @returns {Promise<{
+ *   rootCid: string
+ *   proofSetId: BigInt
+ *   rootId: BigInt
+ *   clientAddress: string
+ * }>}
+ *   The CommP CID of the file.
+ */
+async function getMostRecentFileWithCDN({
+  pdpVerifier,
+  pandoraService
+}) {
+  for (
+    let proofSetId = (await pdpVerifier.getNextProofSetId()) - 1n;
+    proofSetId >= 0n;
```

Comment:
We should stop at `FROM_PROOFSET_ID`. ProofSet older than that may not work, and the original loop never tests them.

Since we are very unlikely to encounter this edge case, I don't mind keeping your current version that iterates all the way down to the proof set id 0.

Reply:
2810173

Reply:
I decided to add it for symmetry and to prevent surprises